### PR TITLE
added views

### PR DESCRIPTION
**Pull request by daniel-waiguru**

### CODE REVIEW

**Comment by abunur** [comment]
Is this your PR for Week 3 homework?

**Comment by daniel-waiguru** [comment]
Yes sir

**Comment by daniel-waiguru** [comment]
yes it is
should i merge it to master first?
sorry for the confusion

**Comment by daniel-waiguru** [comment]
i should merge the previous week's branch to master and close the PR before
creating a new branch for this week right?

**Comment by abunur** [comment]
You can merge, or not - when you merge it is up to you. However my advice is, when you do merge your PR branches, merge them in order. i.e. merge week 3 before week 4, and so on. This will help avoid conflicts. Eventually you will want to merge your PRs so your master branch gets up to speed.

**Comment by daniel-waiguru** [comment]
Thanks
but should i merge before or after you  review ?

**Comment by abunur** [comment]
Merge after review, always. Otherwise, I won't have an open PR to review, anymore.

File: Week03/BusinessCard/app/src/main/java/tech/danielwaiguru/businesscard/MainActivity.kt [hunk 1]
```diff
@@ -1,11 +1,95 @@
...
+        "onPause() - called by the system to indicate that the user is leaving the fragment",
+        "onStop() - stops the fragment",
+        "onDestroyView() - contains the logic UI specific things that need to be cleaned up",
+        "onDestroy() - does the final clean up of the fragment's state",
+        "onDetach() - notify that the fragment has been disassociated from its hosting activity"
+    )
+    companion object{
+        private const val TIP_KEY = "TIP KEY"
+    }
+    lateinit var currentTip: String
```
Comment (abunur) [inline Week03/BusinessCard/app/src/main/java/tech/danielwaiguru/businesscard/MainActivity.kt]: pay attention to warnings in the IDE. Here it's advising you to scope this var as private

**Comment by abunur** [comment]
Nice job overall! Love your use of animations, the menu, and separate layouts for portrait and orientation mode.

Be sure to test your app fully in the future. When I rotate the phone, the app crashes, which is not the best user experience.  Debug your app and have a look at the warnings and issues that come up in the debug console.
On screen rotation there is a crash, with the message
> lateinit property currentTip has not been initialized
In this case you are trying to access a value that does not yet exist in your `onSaveInstanceState` override.

Your grade for this assignment is "good job"

**Review by abunur — CHANGES_REQUESTED** [review]

File: Week03/BusinessCard/app/src/main/java/tech/danielwaiguru/businesscard/MainActivity.kt (hunk 1 above)
Comment (daniel-waiguru) [reply Week03/BusinessCard/app/src/main/java/tech/danielwaiguru/businesscard/MainActivity.kt]: Thank you 
i just resolved the crash by adding an else block as below 
``` 
currentTip = if (savedInstanceState != null){
            savedInstanceState.getString(TIP_KEY)!!
        } else {
            androidTips.random()
        }
        txtTips.text = currentTip
```

**Pull request merged**